### PR TITLE
Spl timechart limit improvement

### PR DESCRIPTION
**Pull request by Jiale-Fang**
# Description
Add support for non-numeric func(Values, Cardinality) for timechart limit option.

Timechart will only display N highest/lowest scoring distinct values of the split-by field. For those cols can not be displayed, they will be merged into the `other` category col.

For numeric func, we can use addition to do the merge logic. For non-numeric func, each one has its own merge logic.

So now it support these kinds of queries,
```
* | timechart span=10m dc(latitude) BY http_method limit=bottom 3
* | timechart span=2m values(first_name) BY http_method limit=bottom3
* | timechart span=2m dc(address), values(first_name) BY http_method limit=2
```

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/aggregations/timechartagg.go [hunk 1]
```diff
@@ -255,22 +264,37 @@ func MergeVal(eVal *utils.CValueEnclosure, eValToMerge utils.CValueEnclosure, hl
...
-		// 		err := hll.Merge(hllToMerge)
-		// 		if err != nil {
-		// 			log.Errorf("MergeVal: failed to merge hyperloglog stats: %v", err)
-		// 		}
-		// 		eVal.CVal = hll.Estimate()
-		// 		eVal.Dtype = utils.SS_DT_UNSIGNED_NUM
-		// 		return
-		// 	}
-		// case utils.Values:
+	// TODO: should merge values for not numeric agg
```
Comment (AndrewHess) [inline pkg/segment/aggregations/timechartagg.go:267]: This TODO is done, right?

File: pkg/segment/aggregations/timechartagg.go [hunk 2]
```diff
@@ -255,22 +264,37 @@ func MergeVal(eVal *utils.CValueEnclosure, eValToMerge utils.CValueEnclosure, hl
...
+			}
+			eVal.CVal = hll.Estimate()
+			eVal.Dtype = utils.SS_DT_UNSIGNED_NUM
+			return
+		}
+	case utils.Values:
+		// Can not do addition for values func
+		if useAdditionForMerge {
+			return
+		}
```
Comment (AndrewHess) [inline pkg/segment/aggregations/timechartagg.go:284]: Does this need a TODO and/or a log message, or should it just be `return`?

File: pkg/segment/aggregations/timechartagg.go (hunk 1 above)
Comment (Jiale-Fang) [reply pkg/segment/aggregations/timechartagg.go]: Yes

File: pkg/segment/aggregations/timechartagg.go (hunk 2 above)
Comment (Jiale-Fang) [reply pkg/segment/aggregations/timechartagg.go:283]: It should be return. `useAdditionForMerge` means sum up their values. It is used by `limit` to score single agg. However, for single agg `values`, we can not sum up them, we use `frequency` to score them.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**